kvstore_api/README_KVStore_api.md

```diff
@@ -0,0 +1,205 @@
+
+A similar scenario is run on each of the possible KVStore class modules:
+
+- TDBStore - providing static wear-leveling and quick access, when you have a small number of KV pairs.
+- FileSystemStore - providing a key-value store API on top of a POSIX-like file system API.
+- SecureStore - provideing encryption, authentication and rollback protection on top of the KVStore API (requires two KVStore instances: one that provides the storage for the KV pairs and one that provides playback attack protection)
+
+
+
+**Table of contents:**
+
+1. [Usage](#usage)
+   - [Import the example](#import-the-example)
+   - [Compile the example](#compile-the-example)
+   - [Run the example](#run-the-example)
+   - [Troubleshooting](#troubleshooting)
+
+## Usage
+
+#### Import the example
+
+Make sure you have an Mbed development environment set up. [Get started with Mbed OS](https://os.mbed.com/docs/latest/tutorials/your-first-program.html)
+to set everything up.
+
+From the command-line, import the example:
+
+```
+mbed import mbed-os-example-kvstore
+cd mbed-os-example-kvstore/kvstore_api
+```
+
+#### Compile the example
+
+Invoke `mbed compile`, and specify the name of your platform and your favorite
+toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the ARM Compiler 5:
+
+```
+mbed compile -m K64F -t ARM
+```
+
+Your PC may take a few moments to compile your code. At the end, you'll see a
+result similar to:
+
+```
+[snip]
++--------------------------+-------+-------+-------+
+| Module                   | .text | .data |  .bss |
++--------------------------+-------+-------+-------+
+| [fill]                   |   235 |     4 |   345 |
+| [lib]/c.a                | 31247 |  2472 |    89 |
+| [lib]/gcc.a              |  3112 |     0 |     0 |
+| [lib]/misc               |   204 |     4 |    28 |
+| [lib]/nosys.a            |    32 |     0 |     0 |
+| main.o                   |  1719 |     0 |     0 |
+| mbed-os/cmsis            |  1029 |     0 |    84 |
+| mbed-os/components       |  4161 |     0 |    48 |
+| mbed-os/drivers          |  1487 |     0 |   100 |
+| mbed-os/features         | 42061 |     4 |  1426 |
+| mbed-os/hal              |  2043 |     4 |    68 |
+| mbed-os/platform         |  3095 |   260 |   134 |
+| mbed-os/rtos             |  7804 |   168 |  5969 |
+| mbed-os/targets          | 15561 |    12 |   413 |
+| Subtotals                |113790 |  2928 |  8704 |
++--------------------------+-------+-------+-------+
+Total Static RAM memory (data + bss): 11632(+0) bytes
+Total Flash memory (text + data): 116718(+0) bytes
+
+Image: ./BUILD/K82F/GCC_ARM/kvstore_api.bin
+```
+
+#### Run the example
+
+1. Connect your Mbed Enabled device to the computer over USB.
+1. Copy the binary file to the Mbed Enabled device.
+1. Press the reset button to start the program.
+1. Open the UART of the board in your favorite UART viewing program. For
+   example, `screen /dev/ttyACM0`.
+
+**Note:** The default serial port baud rate is 9600 bit/s.
```

Comment:
Consider changing this to 115200, as you already have a config file where you can do it (you can look at the NVStore example).